### ISSUE TITLE
jade is now called pug  (trademark problem)-- we should probably fix it so flycheck works

### ISSUE THREAD

**Issue by robbyoconnor** [flycheck/flycheck#951]
I'm gonna take this on but is there a way to handle this in a way that doesn't break the current jade support?

**Comment by cpitclaudel** [comment]
I take it you're referring to https://github.com/pugjs/pug/issues/2184?

@lunaryorn will confirm, but I think we can just add a separate `pug` checker to Flycheck with the same definition as the jade one (but predicated on `pug-mode` — I imagine that's what the Emacs mode will be renamed to). Once the `pug→jade` transition is complete we'll deprecate the jade checker.

We generally don't have too much compatibility stuff like this, but I'd vote for making an exception on this one given the exceptionally annoying process the Jade people are being forced to go through for silly reasons.

**Comment by lunaryorn** [comment]
@robbyoconnor Thank you for making us aware of this issue, but please provide more background information.  We simply cannot follow all developments in all the languages that we support.

What is the current status of jade?  Is it renamed already, with a proper release, or is the rename still ongoing?  In how far does the renaming affect us other than that we need to rename the syntax checker?  In how far would this break the current support for Jade?  Does the renamed version also behave differently?

**Comment by robbyoconnor** [comment]
@lunaryorn -- as of right now the domain name is registered but the site is very much in progress-- the github org is migrated -- but the emacs major-mode is currently still `jade-mode`  -- This doesn't look like it's gonna be a lot of work to do -- it's looking like more or less they just did a name change removing all traces of `Jade/jade`.

**Comment by lunaryorn** [comment]
@robbyoconnor I'm sorry but I'm still not sure in how far this affects us, nor why it could break Flycheck.

Presuming that `pug` still has the same CLI interface as `jade`, wouldn't it be sufficient to just rename the syntax checker to `pug`, make it run `pug` instead of `jade` and add a  (yet to come?) `pug-mode` to its `:modes`?  

That's something we could do as soon as the tool is officially renamed, though not before, to avoid confusion.

**Comment by robbyoconnor** [comment]
That is one option but there is still code out there that's running jade -- or completely unaware this happened

**Comment by lunaryorn** [comment]
@robbyoconnor What's the status of this?

**Comment by robbyoconnor** [comment]
I was waiting to hear back on what I should do...supporting both sounds
like a lot of work...maybe I'll just support pug?

--Rob

**Comment by lunaryorn** [comment]
Ah sorry.  Yes, well only support pug imho.  Let's keep things simple ☺️ 

Is the rename complete already?

**Comment by robbyoconnor** [comment]
I believe so -- give me a couple day? I'll reference this issue in my PR

**Comment by cpitclaudel** [comment]
@robbyoconnor Perfect, thanks!

**Comment by lunaryorn** [comment]
@robbyoconnor Any news here?  Did you get a chance to work on renaming Jade to Pug in Flycheck?

**Comment by nicolasmccurdy** [comment]
Pug 2 was released in August, which I believe is the first official renamed release. I'm also interesting in progress on this, does anyone want help?

**Comment by robbyoconnor** [comment]
I'll do this tonight. Sorry folks -- resigned from a role that sucked the life out of me -- gonna move on this! Sorry all.

**Comment by nicolasmccurdy** [comment]
No rush. :) I was just curious if you guys were aware of the release, I think it was a prerelease for a while.

**Comment by cpitclaudel** [comment]
Thanks @robbyoconnor :) And good luck with work! No apologies needed.

**Comment by robbyoconnor** [comment]
:heart:

**Comment by robbyoconnor** [comment]
Still testing, will make a PR after I'm sure this works. We should also look into adding [pug-lint](https://github.com/pugjs/pug-lint) support, I'm willing to take this on in a separate PR.

**Comment by robbyoconnor** [comment]
Okay so there's a backwards compatibility issue. Should we keep the jade checker? The pattern changed, ever so slightly but it's enough to break the jade checker.

**Comment by robbyoconnor** [comment]
I could actually write this in a way that works for Jade/Pug at the same time!

**Comment by lunaryorn** [comment]
@robbyoconnor That'd be the very best option—please let's continue the discussion over in #1084 ☺️
